build.gradle.kts

```diff
@@ -1,105 +1,96 @@
 import org.hypertrace.gradle.publishing.License.APACHE_2_0;
 
 plugins {
-    `java-library`
-    id("com.diffplug.spotless") version "5.2.0" apply false
-    id("org.hypertrace.publish-maven-central-plugin") version "1.0.4" apply false
-    id("org.hypertrace.ci-utils-plugin") version "0.3.0"
-    id("org.gradle.test-retry") version "1.2.0" apply false
-    id("org.owasp.dependencycheck") version "7.1.1"
+  id("com.diffplug.spotless") version "5.2.0" apply false
+  id("org.hypertrace.publish-maven-central-plugin") version "1.0.6"
+  id("org.hypertrace.ci-utils-plugin") version "0.3.0"
+  id("org.gradle.test-retry") version "1.2.0" apply false
+  id("org.owasp.dependencycheck") version "7.1.1"
 }
 
-buildscript {
-    repositories {
-        mavenCentral()
+subprojects {
+  group = "org.hypertrace.agent"
+  description = "Hypertrace OpenTelemetry Javaagent"
+
+  apply(plugin = "java-library")
+  apply(plugin = "com.diffplug.spotless")
+  apply(from = "$rootDir/gradle/spotless.gradle")
+  pluginManager.withPlugin("java-library") {
+    configure<JavaPluginExtension> {
+      sourceCompatibility = JavaVersion.VERSION_1_8
+      targetCompatibility = JavaVersion.VERSION_1_8
     }
-}
 
-allprojects {
-    apply(plugin="java-library")
-    java {
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+    repositories {
+      mavenCentral()
+      maven {
+        url = uri("https://oss.sonatype.org/content/repositories/snapshots/")
+      }
     }
 
-    repositories {
-        mavenCentral()
+    configurations {
```

Comment:
What's the purpose of this block? Where are we including a gradle plugin in the dependency classpath for example?